hooks/settings.py

```diff
@@ -0,0 +1,187 @@
+         </provider>
+      </essence>
+   </format>
+</codecprofile>
+        """
+        
+        path = self._write_content_to_file(xml, "quicktime_settings.cdxprof")
+        return path
+    
+    def _generate_profile_xml(self):
+        """
+        Generate flame export profile settings suitable for generating a single quicktime 
+        file to represent an entire sequence.
+        
+        :returns: path to export preset xml file
+        """
+        
+        # this profile is a 8-bit QuickTime file (H.264 1280x720 8Mbits) with 
+        # preset High_Blu_Ray_8Mbits, suitable for upload to Shotgun Review.
+        #
+        # for more details around quicktime settings and shotgun, see
+        # https://support.shotgunsoftware.com/entries/26303513-Transcoding
+        
+        xml = """<?xml version="1.0" encoding="UTF-8"?>
+<preset version="4">
+   <type>movie</type>
+   <comment>Creates an 8-bit QuickTime file (H.264 1280x720 8Mbits).</comment>
+   <movie>
+      <fileType>QuickTime</fileType>
+      <namePattern></namePattern>
+      <yuvHeadroom>False</yuvHeadroom>
+      <yuvColourSpace>PCS_UNKNOWN</yuvColourSpace>
+      <operationalPattern>None</operationalPattern>
+      <companyName>Autodesk</companyName>
+      <productName />
+      <versionName />
+   </movie>
+   <video>
+      <fileType>QuickTime</fileType>
+      <codec>33622016</codec>
+      <codecProfile>{CODE_PROFILE_PATH}</codecProfile>
+      <namePattern></namePattern>
+      <compressionQuality>50</compressionQuality>
+      <transferCharacteristic>2</transferCharacteristic>
+      <colorimetricSpecification>4</colorimetricSpecification>
+      <publishLinked>False</publishLinked>
+      <foregroundPublish>False</foregroundPublish>
+      <overwriteWithVersions>False</overwriteWithVersions>
+      <resize>
+         <resizeType>fit</resizeType>
+         <resizeFilter>lanczos</resizeFilter>
+         <width>1280</width>
+         <height>720</height>
+         <bitsPerChannel>8</bitsPerChannel>
+         <numChannels>3</numChannels>
+         <floatingPoint>False</floatingPoint>
+         <bigEndian>True</bigEndian>
+         <pixelRatio>1</pixelRatio>
+         <scanFormat>P</scanFormat>
+      </resize>
+   </video>
+   <audio>
+      <fileType>QuickTime</fileType>
+      <codec>4026601474</codec>
+      <codecProfile />
+      <namePattern></namePattern>
+      <mixdown>AsIs</mixdown>
+      <sampleRate>48000</sampleRate>
+      <bitRate>128</bitRate>
+      <bitDepth>-1</bitDepth>
+   </audio>
+   <name>
+      <framePadding>8</framePadding>
+      <startFrame>0</startFrame>
+      <useTimecode>False</useTimecode>
+   </name>
+</preset>
+        """
+        # first generate the quicktime presets and bind this up to the content above
+        quicktime_settings_path = self._generate_quicktime_settings()
+        # plug in the path to the quicktime preset
+        resolved_xml = xml.replace("{CODE_PROFILE_PATH}", quicktime_settings_path)
```

Comment:
Why not using a standard string formatter, like %s ? What if other values need to be set before writing the data ?